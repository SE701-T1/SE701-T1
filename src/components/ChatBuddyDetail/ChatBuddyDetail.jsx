import React from 'react';
import { Button, Box, Typography } from '@mui/material';
import PropTypes from 'prop-types';
import styles from './ChatBuddyDetail.module.css';

import ProfileBadge from '../ProfileBadge/ProfileBadge';

/**
 * ChatBuddyDetail component displays the name of the buddy and the courses they share.
 * commonCourses is an array of objects with courseName and courseLink properties.
 * Example:
 *  {
 *    courseName: 'SOFTENG 701',
 *    courseLink: 'https://random.link',
 *  }
 */
function ChatBuddyDetail({ self, active, name, removeBuddy, commonCourses }) {
  return (
    <Box className={styles.container}>
      <ProfileBadge active={active} name={name} />
      <Typography fontWeight={600} variant="h5">
        {name}
      </Typography>
      {active && <Typography className={styles.activeText}> Active now</Typography>}
      {!self && (
        <Button
          className={styles.removeButton}
          color="primaryDark"
          variant="outlined"
          data-testid="remove-buddy"
          onClick={removeBuddy}>
          Remove Buddy
        </Button>
      )}
      <Typography className={styles.commonCourseTitle}>
        {self ? 'My Courses' : 'Common Courses'}
      </Typography>
      {commonCourses.map(({ courseName, courseLink }) => (
        <Typography className={styles.commonCourseText} key={courseName + courseLink}>
          {courseName}
        </Typography>
      ))}
    </Box>
  );
}

ChatBuddyDetail.propTypes = {
  active: PropTypes.bool,
  name: PropTypes.string.isRequired,
  removeBuddy: PropTypes.func.isRequired,
  commonCourses: PropTypes.arrayOf(
    PropTypes.shape({
      courseName: PropTypes.string.isRequired,
      courseLink: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

ChatBuddyDetail.defaultProps = {
  active: false,
};

export default ChatBuddyDetail;
