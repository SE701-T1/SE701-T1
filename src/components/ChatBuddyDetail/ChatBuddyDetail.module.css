.container {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.activeText {
  color: #8896ac;
  font-weight: 500;
}

.removeButton {
  text-transform: none !important;
  margin-top: 50px !important;
  display: flex;
  align-items: center;
  text-align: center;
  padding: 5px 20px 5px 20px;
}

.commonCourseTitle {
  margin-top: 30px !important;
  margin-bottom: 5px !important;
  color: #8896ac;
  font-weight: 600;
}

.commonCourseText {
  text-decoration: underline;
}
