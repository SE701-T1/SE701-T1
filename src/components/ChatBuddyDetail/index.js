import ChatBuddyDetail from './ChatBuddyDetail';

export default ChatBuddyDetail;
