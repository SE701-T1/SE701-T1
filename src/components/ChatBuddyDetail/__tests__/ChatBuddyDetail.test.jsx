import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/material';
import theme from '../../Theme';

import ChatBuddyDetail from '../ChatBuddyDetail';

const mockCourse = [
  {
    courseName: 'SOFTENG 701',
    courseLink: 'https://random.link',
  },
];

describe('ChatBuddyDetail', () => {
  it('should display member', () => {
    render(
      <ThemeProvider theme={theme}>
        <ChatBuddyDetail name="test name" removeBuddy={() => {}} commonCourses={mockCourse} />
      </ThemeProvider>,
    );

    expect(screen.getByText('test name')).toBeInTheDocument();
    expect(screen.getByText('SOFTENG 701')).toBeInTheDocument();
  });

  it('should display active', () => {
    render(
      <ThemeProvider theme={theme}>
        <ChatBuddyDetail
          active
          name="test name"
          removeBuddy={() => {}}
          commonCourses={mockCourse}
        />
      </ThemeProvider>,
    );

    expect(screen.getByText('test name')).toBeInTheDocument();
  });

  it('should removeBuddy when pressed', () => {
    const mockRemoveBuddy = jest.fn();

    render(
      <ThemeProvider theme={theme}>
        <ChatBuddyDetail
          name="test name"
          removeBuddy={mockRemoveBuddy}
          commonCourses={mockCourse}
        />
      </ThemeProvider>,
    );

    fireEvent.click(screen.getByTestId('remove-buddy'));
    expect(mockRemoveBuddy).toHaveBeenCalledTimes(1);
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ChatBuddyDetail name="test name" removeBuddy={() => {}} commonCourses={mockCourse} />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
