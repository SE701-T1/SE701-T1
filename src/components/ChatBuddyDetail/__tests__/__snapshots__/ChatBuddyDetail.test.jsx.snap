// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container MuiBox-root css-0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-6ch1n3-MuiContainer-root"
  >
    <span
      className="MuiBadge-root css-wd6a4w-MuiBadge-root"
      data-testid="offline"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1pl8iwh-MuiAvatar-root"
        data-testid="test name"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <span
        className="MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight css-ank3r8-MuiBadge-badge"
      >
        
      </span>
    </span>
  </div>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-1xxajqc-MuiTypography-root"
  >
    test name
  </h5>
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimaryDark MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root removeButton css-kssayf-MuiButtonBase-root-MuiButton-root"
    data-testid="remove-buddy"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Remove Buddy
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1 commonCourseTitle css-ahj2mt-MuiTypography-root"
  >
    Common Courses
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 commonCourseText css-ahj2mt-MuiTypography-root"
  >
    SOFTENG 701
  </p>
</div>
`;
